minetest/minetest
# core.clear_objects function overwrite to add more flexibility

This extends the possibilities of command `/clearobjects` and Lua function `core.clear_objects`.

- This is related to #6124
- Add two new mods to the `clear_objects` function. 

## To do

This PR is a Work in Progress.

- [ ] Apply feedback
- [ ] Document result.

## How to test

Run devtest game, add the entity, and try command `/clearobjects`.

For mode `rules` check object by luaentity table use /clearobjects rules {e={...}} .
For mode `rules` check object by properties table use /clearobjects rules {p={...}} .

So use etc `/clearobjects rules {p={infotext="INFo",nametag="TAG"}}`  and only objects where you manually edit props to match these values, will be deleted.

For mode `soft`, only entities with luaentity table value `prevent_soft_clearobjects` should be kept alive.
This can be nicely used by mods like mob_animals to prevent the clearing of tamed ones. `soft` mode is now the default mode for `/clearobejcts` command.
